src/components/UI/AvailabilityZonesLabel.js

```diff
@@ -104,9 +104,17 @@ function AvailabilityZonesLabel({
   const color = azColors[colorIndex];
 
   const toggleChecked = () => {
-    if (!isMaxReached || isChecked || isRadioButtons) {
-      onToggleChecked(!isChecked, { title, value, label });
+    // This component is used in forms, where we do pass an onToggleChecked function
+    // to change form state, and in cluster details view where we do not need to
+    // perform any action onClick.
+    // We only want to return onToggleChecked if it is passed.
+    if ((!isMaxReached || isChecked || isRadioButtons) && onToggleChecked) {
+      return onToggleChecked(!isChecked, { title, value, label });
```

Comment:
You could get rid of that `&& onToggleChecked` and use `onToggleChecked?.(...)` 👍

Reply:
Didn't know about this! I will leave it as I have it now, ESLint is yelling at me

Reply:
Unfortunatelly ESLint is still stupid about optional chaining, but you can disable the rule for the line if you want to use it 👍

Reply:
Niice, thanks for keeping me updated about this new stuff! :-)